src/driving/vehicle_boat.py

```diff
@@ -0,0 +1,276 @@
+
+        else:
+            self.wind_assist = 1 + (91 - abs(270 - vehicle_angle)) / 45
+
+    def apply_reductive_force(self, force: float, delta_time: float):
+        # frictional forces or brake forces should reduce any acting velocity
+        #  to zero but not into negative the way reverse acceleration would
+        reductive_force = force * delta_time
+        x = self.clamp(abs(self.external_velocity[0]) - reductive_force, 0, 999)
+        y = self.clamp(abs(self.external_velocity[1]) - reductive_force, 0, 999)
+        angle = self.clamp(abs(self.external_velocity[2]) - reductive_force, 0, 999)
+
+        x *= math.copysign(1, self.external_velocity[0])
+        y *= math.copysign(1, self.external_velocity[1])
+        angle *= math.copysign(1, self.external_velocity[2])
+
+        self.external_velocity = (x, y, angle)
+
+    def get_real_velocity(self, delta_time):
+
+        # get the force from external sources for this frame
+        x = self.external_velocity[0] * delta_time
+        y = self.external_velocity[1] * delta_time
+        angle = self.external_velocity[2] * delta_time
+        # then if not in the lockout window from a collision
+        # get the intended force from the vehicle
+        if self.in_collision is False:
+            x += self.vehicle_velocity[0] * delta_time
+            y += self.vehicle_velocity[1] * delta_time
+            # the boat can turn while stationary
+            angle += self.vehicle_velocity[2] * delta_time
+
+        # if the boat is facing with the 'fake' wind direction get bonus speed
+        x *= self.wind_assist
+        y *= self.wind_assist
+
+        # if self.vehicle_velocity[0] != 0:
+        #     print(self.wind_assist)
+        # if the anchor is down the boat cant move but CAN turn
+        if self.anchor_position > 0.9:
+            return (0, 0, angle)
+
+        return (x, y, angle)
+
+    def apply_external_force(self, vector: (float, float, float)):
+        x = vector[0] + self.external_velocity[0]
+        y = vector[1] + self.external_velocity[1]
+        angle = vector[2] + self.external_velocity[2]
+        self.external_velocity = (x, y, angle)
+
+    def collide_at_point(self, point, velocity, delta_time):
+
+        if self.in_collision is False and self.impact_buffer == 0:
+            self.in_collision = True
+            self.impact_buffer = DEFAULT_CONTROL_LOCKOUT
+
+            collision_vector = (
+                -velocity[0] * 0.9 / delta_time,
+                -velocity[1] * 0.9 / delta_time,
+                -velocity[2] * 0.9 / delta_time,
+            )
+
+            # collisions should over ride any external forces on the car
+            # since those forces are included in the calculation of the collision vector
+            self.external_velocity = collision_vector
+            self.current_acceleration = (0, 0)
+
+    def clamp(self, num, min_value, max_value):
+        return max(min(num, max_value), min_value)
+
+    def lerp(
```

Comment:
can you suggest a video or other primer on using lerp?  I understand it's industry standard term, but it's opaque to me